### PR TITLE
Initial C11 libatomic GCC builtin implementation

### PR DESCRIPTION
**Pull request by gyrovorbis**
This PR is to add the remaining GCC builtins required to enable all of C11's atomics, located in `<stdatomic.h>`. The majority of the builtins for primitive types are given to us by the compiler with `-matomic-model=soft-imask`; however, this evidently does not cover 64-bit (long long) primitive types nor does it cover the generic cases for atomic structures and buffers.

I've implemented these missing builtins, and I've implemented the 64-bit primitive back-ends using generic macros to generate the function calls, just in case we find anything else missing or wind up wanting to handle all primitive atomics by ourselves. 

- Initial implementation should cover everything C11 atomics entail, located at kernel/libc/c11/atomics.c

All of my preliminary tests are passing; however, I'm keeping this as a draft until I've created a KOS example/test scenario fully exercising all of C11 atomics.

### CODE REVIEW

**Comment by gyrovorbis** [comment]
@ljsebald I was pondering some stuff last night and wanted to double check something with you real quick.

The C11 atomics API lets you query the implementation via `atomic_is_lock_free()` on an atomic to determine whether it "locks" or not. The default implementation for the stuff GCC is giving us via `-matomic-model=soft-imask` is all returning `true` there, so that model is clearly considered to be lock-free.

The question, though, is now that we're using spinlocks for the generic-case, is that still considered to be a "lock-free" implementation by the C standard, since it's technically non-blocking or do I need to modify that symbol to return `false` for the generic case?

**Comment by ljsebald** [comment]
> @ljsebald I was pondering some stuff last night and wanted to double check something with you real quick.
> 
> The C11 atomics API lets you query the implementation via `atomic_is_lock_free()` on an atomic to determine whether it "locks" or not at runtime. The default implementation for the stuff GCC is giving us via `-matomic-model=soft-imask` is all returning `true` there, so that model is clearly considered to be lock-free.
> 
> The question, though, is now that we're using spinlocks for the generic-case, is that still considered to be a "lock-free" implementation by the C standard, since it's technically non-blocking or do I need to modify that symbol to return `false` for the generic case?

I'd imagine it wouldn't be proper to say that our implementation is lock free when it isn't. In the grand scheme of things, does it matter? Probably not... But to be proper, it should probably be the case that for any object that would trigger the spinlock case that it should say false for it being lock-free.

**Comment by gyrovorbis** [comment]
> > @ljsebald I was pondering some stuff last night and wanted to double check something with you real quick.
> > The C11 atomics API lets you query the implementation via `atomic_is_lock_free()` on an atomic to determine whether it "locks" or not at runtime. The default implementation for the stuff GCC is giving us via `-matomic-model=soft-imask` is all returning `true` there, so that model is clearly considered to be lock-free.
> > The question, though, is now that we're using spinlocks for the generic-case, is that still considered to be a "lock-free" implementation by the C standard, since it's technically non-blocking or do I need to modify that symbol to return `false` for the generic case?
> 
> I'd imagine it wouldn't be proper to say that our implementation is lock free when it isn't. In the grand scheme of things, does it matter? Probably not... But to be proper, it should probably be the case that for any object that would trigger the spinlock case that it should say false for it being lock-free.

Sounds good. I am definitely wanting to do the right thing here.

1) updated back-end and example so that `atomic_is_lock_free()` returns false for our generic case
2) merged back with master
3) fixed a teensy little warning that I must've missed
4) reverified tests, nothing broke

File: kernel/libc/c11/atomics.c [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+ATOMIC_FETCH_N_(unsigned long long, 8, xor, ^=)
+ATOMIC_FETCH_NAND_N_(unsigned long long, 8)
+
+/* Provide GCC with symbols and logic required to implement 
+   generically sized atomics. Rather than disabling an enabling 
+   interrupts around primitive assignments, we use spinlocks 
+   around memcpy() calls. */
+
+/* Size of each memory region covered by an individual lock. */
+#define GENERIC_LOCK_BLOCK_SIZE     (CPU_CACHE_BLOCK_SIZE * 4)
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:123]:
```suggestion
#define GENERIC_LOCK_BLOCK_SIZE     (CPU_CACHE_BLOCK_SIZE * 4)

```

File: kernel/libc/c11/atomics.c [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+inline static uintptr_t 
+address_to_spinlock(const volatile void *ptr) {
+    return ((uintptr_t)ptr / GENERIC_LOCK_BLOCK_SIZE) % GENERIC_LOCK_COUNT;
+}
+
+void __atomic_load(size_t size, 
+                   const volatile void *ptr, 
+                   void *ret, 
+                   int memorder)
+{
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:144]:
```suggestion
                   int memorder) {
```

File: kernel/libc/c11/atomics.c [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    memcpy(ret, (const void *)ptr, size);
+    
+    spinlock_unlock(&locks[lock]);
+}
+
+void __atomic_store(size_t size, 
+                    volatile void *ptr, 
+                    void *val, 
+                    int memorder) 
+{
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:159]:
```suggestion
                    int memorder) {
```

File: kernel/libc/c11/atomics.c [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    
+    spinlock_unlock(&locks[lock]);
+}
+
+void __atomic_exchange(size_t size,
+                       volatile void *ptr,
+                       void *val,
+                       void *ret,
+                       int memorder)
+{
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:175]:
```suggestion
                       int memorder) {
```

File: kernel/libc/c11/atomics.c [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    spinlock_unlock(&locks[lock]);                       
+}
+
+bool __atomic_compare_exchange(size_t size,
+                               volatile void* ptr,
+                               void* expected,
+                               void* desired,
+                               int success_memorder,
+                               int fail_memorder)
+{
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:193]:
```suggestion
                               int fail_memorder) {
```

File: kernel/libc/c11/atomics.c [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    (void)success_memorder;
+    (void)fail_memorder;
+
+    bool retval;
+    const uintptr_t lock = address_to_spinlock(ptr);
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:198]:
```suggestion
    bool retval;
    const uintptr_t lock = address_to_spinlock(ptr);

    (void)success_memorder;
    (void)fail_memorder;
```

File: kernel/libc/c11/atomics.c [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+    
+    spinlock_lock(&locks[lock]);
+    
+    if(memcmp((const void *)ptr, expected, size) == 0) {
+        memcpy((void *)ptr, desired, size);
+        retval = true;
+    } else {
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:205]:
```suggestion
    }
    else {
```

File: kernel/libc/c11/atomics.c [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+
+void __atomic_exchange(size_t size,
+                       volatile void *ptr,
+                       void *val,
+                       void *ret,
+                       int memorder)
+{
+    (void)memorder;
+
+    const uintptr_t lock = address_to_spinlock(ptr);
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:178]:
```suggestion
    const uintptr_t lock = address_to_spinlock(ptr);

    (void)memorder;

```

File: kernel/libc/c11/atomics.c [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+void __atomic_store(size_t size, 
+                    volatile void *ptr, 
+                    void *val, 
+                    int memorder) 
+{
+    (void)memorder;
+    
+    const uintptr_t lock = address_to_spinlock(ptr);
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:162]:
```suggestion
    const uintptr_t lock = address_to_spinlock(ptr);

    (void)memorder;
    
```

File: kernel/libc/c11/atomics.c [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+void __atomic_load(size_t size, 
+                   const volatile void *ptr, 
+                   void *ret, 
+                   int memorder)
+{
+    (void)memorder;
+
+    const uintptr_t lock = address_to_spinlock(ptr);
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:147]:
```suggestion
    const uintptr_t lock = address_to_spinlock(ptr);

    (void)memorder;

```

File: kernel/libc/c11/atomics.c [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+#define ATOMIC_FETCH_NAND_N_(type, n) \
+    type \
+    __atomic_fetch_nand_##n(volatile void* ptr, \
+                            type val, \
+                            int memorder) { \
+        (void)memorder; \
+        const int irq = irq_disable(); \
+        type ret = *(type *)ptr; \
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:99]:
```suggestion
        const int irq = irq_disable(); \
        type ret = *(type *)ptr; \
        (void)memorder; \
```

File: kernel/libc/c11/atomics.c [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+#define ATOMIC_FETCH_N_(type, n, opname, op) \
+    type \
+    __atomic_fetch_##opname##_##n(volatile void* ptr, \
+                                  type val, \
+                                  int memorder) { \
+        (void)memorder; \
+        const int irq = irq_disable(); \
+        type ret = *(type *)ptr; \
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:86]:
```suggestion
        const int irq = irq_disable(); \
        type ret = *(type *)ptr; \
        (void)memorder; \
```

File: kernel/libc/c11/atomics.c [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+                                  type desired, \
+                                  bool weak, \
+                                  int success_memorder, \
+                                  int failure_memorder) \
+    { \
+        (void)weak; \
+        (void)success_memorder; \
+        (void)failure_memorder; \
+        const int irq = irq_disable(); \
+        bool retval; \
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:67]:
```suggestion
                                  int failure_memorder) { \
        const int irq = irq_disable(); \
        bool retval; \
        (void)weak; \
        (void)success_memorder; \
        (void)failure_memorder; \
```

File: kernel/libc/c11/atomics.c [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+        *(type *)ptr = val; \
+        irq_restore(irq); \
+    }
+    
+#define ATOMIC_EXCHANGE_N_(type, n) \
+    type \
+    __atomic_exchange_##n(volatile void* ptr, type val, int model) { \
+        (void)model; \
+        const int irq = irq_disable(); \
+        const type ret = *(type *)ptr; \
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:48]:
```suggestion
        const int irq = irq_disable(); \
        const type ret = *(type *)ptr; \
        (void)model; \
```

File: kernel/libc/c11/atomics.c [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+/* Create a set of macros to codegen atomic symbols for primitive types. 
+   For these types, we simply disable interrupts then reenable them 
+   around accesses to our atomics to ensure their atomicity.
+*/
+#define ATOMIC_LOAD_N_(type, n) \
+    type \
+    __atomic_load_##n(const volatile void *ptr, int model) { \
+        (void)model; \
+        const int irq = irq_disable(); \
+        const type ret = *(type *)ptr; \
```
Comment (ljsebald) [inline kernel/libc/c11/atomics.c:29]:
```suggestion
        const int irq = irq_disable(); \
        const type ret = *(type *)ptr; \
        (void)model; \
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 16]
```diff
@@ -0,0 +1,388 @@
...
+}
+
+/* Per-thread entry point. */
+int thread(void *arg) { 
+   unsigned tid = (unsigned)arg;
+   int retval = 0;
+
+   /* Do several iterations worth of atomic operations for each thread. */
+   for(unsigned i = 0; i < ITERATION_COUNT; ++i) {
+
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:118]:
```suggestion
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 17]
```diff
@@ -0,0 +1,388 @@
...
+      /* Create a second type of spinlock out of an atomic 
+         boolean, doing the same thing as before: attempting
+         to set "bool_atomic" to true if it was previously false
+         to claim the lock. */
+      bool expected = false;
+      while(!atomic_compare_exchange_weak(&bool_atomic,
+                                          &expected,
+                                          true))
+      {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:155]:
```suggestion
                                          true)) { 
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 18]
```diff
@@ -0,0 +1,388 @@
...
+      while(!atomic_compare_exchange_weak(&bool_atomic,
+                                          &expected,
+                                          true))
+      { 
+         printf("Thread[%2u]: Waiting to acquire atomic bool lock.\n", tid);
+         /* This time we won't hog all of the CPU time waiting. */
+         thrd_yield();
+         expected = false;     
+      }
+      printf("Thread[%2u]: Acquired atomic bool lock.\n", tid);
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:161]:
```suggestion

      printf("Thread[%2u]: Acquired atomic bool lock.\n", tid);
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 19]
```diff
@@ -0,0 +1,388 @@
...
+      printf("Thread[%2u]: Releasing atomic bool lock.\n", tid);
+      
+      /* Do the reverse compare + exchange operation to release the lock, this
+         time expecting it to be previously owned (true) and setting it to 
+         false to release. */
+      expected = true;
+      if(!atomic_compare_exchange_strong(&bool_atomic,
+                                         &expected,
+                                         false))
+      {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:180]:
```suggestion
                                         false)) {
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 20]
```diff
@@ -0,0 +1,388 @@
...
+   }
+
+   printf("Running threads: [%u]\n", THREAD_COUNT);
+
+   /* Create N-1 threads running the "thread()" function to
+      manipulate and fight over various atomic data. */ 
+   for(unsigned t = 0; t < THREAD_COUNT - 1; ++t) {
+      if(thrd_create(&threads[t], thread, (void *)t + 1)
+            != thrd_success) 
+      {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:251]:
```suggestion
      if(thrd_create(&threads[t], thread,
                     (void *)t + 1) != thrd_success) {
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 21]
```diff
@@ -0,0 +1,388 @@
...
+   }   
+
+   /* Validate that all of our atomics have their expected final values. */
+   printf("\nValidating results:\n");
+
+   /* Verify atomic_flag state. */
+   if(atomic_flag_test_and_set(&flag_atomic)) {
+      fprintf(stderr, "flag_atomic left in unexpected state: [true]\n");
+      retval = -1;
+   } else printf("\tFlag atomics work.\n");
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:284]:
```suggestion
   }
   else {
       printf("\tFlag atomics work.\n");
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 22]
```diff
@@ -0,0 +1,388 @@
...
+
+   /* Verify atomic bool state. */
+   bool expected = false;
+   if(!atomic_compare_exchange_weak(&bool_atomic,
+                                    &expected,
+                                    true))
+   {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:291]:
```suggestion
                                    true)) {
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 23]
```diff
@@ -0,0 +1,388 @@
...
+   } else printf("\tBool atomics work.\n");
+
+   /* Verify atomic byte state. */
+   uint8_t byte_value, byte_expected_value = 0;
+   for(unsigned i = 0; i < THREAD_COUNT; ++i)
+      byte_expected_value |= i;
+
+   if((byte_value = atomic_load(&byte_atomic))
+         != byte_expected_value)
+   {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:303]:
```suggestion
   if((byte_value = atomic_load(&byte_atomic)) != byte_expected_value) {
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 24]
```diff
@@ -0,0 +1,388 @@
...
+      fprintf(stderr, 
+              "byte_atomic left in unexpected state: [%u]\n", 
+              byte_value);
+      retval = -1;
+   } else printf("\t8-bit atomics work.\n");
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:308]:
```suggestion
   }
   else {
       printf("\t8-bit atomics work.\n");
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 25]
```diff
@@ -0,0 +1,388 @@
...
+   short short_value;
+   /* Note that memory order shouldn't do anything for our platform, just testing. */
+   if((short_value = atomic_load(&short_atomic))
+         != -(THREAD_COUNT * ITERATION_COUNT))
+   {
+      fprintf(stderr, 
+              "short_atomic left in unexpected state: [%i]\n", 
+              short_value);
+      retval = -1;
+   } else printf("\t16-bit atomics work.\n");
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:320]:
```suggestion
   }
   else {
       printf("\t16-bit atomics work.\n");
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 26]
```diff
@@ -0,0 +1,388 @@
...
+      int_expected_value &= i;
+   
+   if((int_value = atomic_load(&int_atomic)) 
+         != int_expected_value)
+   {
+      fprintf(stderr, 
+              "int_atomic left in unexpected state: [%d]\n", 
+              int_value);
+      retval = -1;
+   } else printf("\t32-bit atomics work.\n");
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:334]:
```suggestion
   }
   else {
       printf("\t32-bit atomics work.\n");
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 27]
```diff
@@ -0,0 +1,388 @@
...
+   /* Verify atomic long long state. */
+   uint64_t longlong_value;
+   if((longlong_value = atomic_load(&longlong_atomic)) 
+         != THREAD_COUNT * ITERATION_COUNT) 
+   {
+      fprintf(stderr, 
+              "longlong_atomic left in unexpected state: [%llu]\n", 
+              longlong_value);
+      retval = -1;
+   } printf("\t64-bit atomics work.\n");
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:345]:
```suggestion
   }
   else {
       printf("\t64-bit atomics work.\n");
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 28]
```diff
@@ -0,0 +1,388 @@
...
+      ptrdiff_expected_value ^= i;
+   
+   if((ptrdiff_value = atomic_load(&ptrdiff_atomic)) 
+         != ptrdiff_expected_value)
+   {
+      fprintf(stderr, 
+              "ptrdiff_atomic left in unexpected state: [%d]\n", 
+              ptrdiff_value);
+      retval = -1;
+   } else printf("\tptrdiff_t atomics work.\n");
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:359]:
```suggestion
   }
   else {
       printf("\tptrdiff_t atomics work.\n");
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 29]
```diff
@@ -0,0 +1,388 @@
...
+
+   /* Verify atomic bool state. */
+   bool expected = false;
+   if(!atomic_compare_exchange_weak(&bool_atomic,
+                                    &expected,
+                                    true))
+   {
+      fprintf(stderr, "bool_atomic left in unexpected state: [true]\n");
+      retval = -1;
+   } else printf("\tBool atomics work.\n");
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:294]:
```suggestion
   }
   else {
       printf("\tBool atomics work.\n");
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 30]
```diff
@@ -0,0 +1,388 @@
...
+   } else printf("\t16-bit atomics work.\n");
+
+   /* Verify atomic int state. */
+   int int_value, int_expected_value = INT_MAX;
+   for(int i = 0; i < THREAD_COUNT; ++i)
+      int_expected_value &= i;
+   
+   if((int_value = atomic_load(&int_atomic)) 
+         != int_expected_value)
+   {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:329]:
```suggestion
   if((int_value = atomic_load(&int_atomic)) != int_expected_value) {
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 31]
```diff
@@ -0,0 +1,388 @@
...
+              "int_atomic left in unexpected state: [%d]\n", 
+              int_value);
+      retval = -1;
+   } else printf("\t32-bit atomics work.\n");
+
+   /* Verify atomic long long state. */
+   uint64_t longlong_value;
+   if((longlong_value = atomic_load(&longlong_atomic)) 
+         != THREAD_COUNT * ITERATION_COUNT) 
+   {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:340]:
```suggestion
   if((longlong_value = atomic_load(&longlong_atomic)) !=
           THREAD_COUNT * ITERATION_COUNT) {
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 32]
```diff
@@ -0,0 +1,388 @@
...
+              byte_value);
+      retval = -1;
+   } else printf("\t8-bit atomics work.\n");
+
+   /* Verify atomic short state. */
+   short short_value;
+   /* Note that memory order shouldn't do anything for our platform, just testing. */
+   if((short_value = atomic_load(&short_atomic))
+         != -(THREAD_COUNT * ITERATION_COUNT))
+   {
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:315]:
```suggestion
   if((short_value = atomic_load(&short_atomic)) !=
           -(THREAD_COUNT * ITERATION_COUNT)) {
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 33]
```diff
@@ -0,0 +1,388 @@
...
+      fprintf(stderr, 
+              "short_atomic left in unexpected state: [%i]\n", 
+              short_value);
+      retval = -1;
+   } else printf("\t16-bit atomics work.\n");
+
+   /* Verify atomic int state. */
+   int int_value, int_expected_value = INT_MAX;
+   for(int i = 0; i < THREAD_COUNT; ++i)
+      int_expected_value &= i;
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:325]:
```suggestion
   for(int i = 0; i < THREAD_COUNT; ++i) {
       int_expected_value &= i;
   }
```

File: examples/dreamcast/basic/threading/atomics/atomics.c [hunk 34]
```diff
@@ -0,0 +1,388 @@
...
+                                    true))
+   {
+      fprintf(stderr, "bool_atomic left in unexpected state: [true]\n");
+      retval = -1;
+   } else printf("\tBool atomics work.\n");
+
+   /* Verify atomic byte state. */
+   uint8_t byte_value, byte_expected_value = 0;
+   for(unsigned i = 0; i < THREAD_COUNT; ++i)
+      byte_expected_value |= i;
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/atomics/atomics.c:299]:
```suggestion
   for(unsigned i = 0; i < THREAD_COUNT; ++i) {
       byte_expected_value |= i;
   }
```

File: examples/dreamcast/basic/threading/Makefile [hunk 35]
```diff
@@ -1,7 +1,7 @@
 # KallistiOS ##version##
 #
 # examples/dreamcast/basic/threading/Makefile
-# (c)2002 Megan Potter
+# (c) 2002 Megan Potter
```
Comment (ljsebald) [inline examples/dreamcast/basic/threading/Makefile:4]:
```suggestion
# Copyright (C) 2002 Megan Potter
```

**Review by ljsebald — CHANGES_REQUESTED** [review]

**Comment by gyrovorbis** [comment]
Alrighty,  addressed review changes and retested, because I'm always paranoid. Looking good!

Screenshot from 2023-09-15 02-14-39

**Review by ljsebald — APPROVED** [review]

**Pull request merged**